subsys/bluetooth/audio/Kconfig.tbs

```diff
@@ -170,6 +170,171 @@ config BT_TBS_CLIENT_MAX_TBS_INSTANCES
 	  Sets the maximum number of Telephone Bearer Service (TBS)
 	  instances to setup and use.
 
+config BT_TBS_CLIENT_MINIMAL
+	bool "Minimal TBS Client without optional procedures"
+	default n
+	help
+	  This option disables all optional procedures in the TBS Client.
+
+config BT_TBS_CLIENT_BEARER_PROVIDER_NAME
+	bool "Support reading Bearer Provider Name"
+	default y
+	depends on !BT_TBS_CLIENT_MINIMAL
```

Comment:
I think I would prefer the `BT_TBS_CLIENT_MINIMAL` to not be restrictive. 

The use case that I want to cover is that we only want to enable e.g. `BT_TBS_CLIENT_BEARER_PROVIDER_NAME` and `BT_TBS_CLIENT_BEARER_TECHNOLOGY`. 

This can be done in the following ways:
1) `default n` for all optional characteristics, and enable those 2
2) `default y` for all optional characteristics (current implementation), and disable the remaining 18 Kconfigs
3) `default y` for all optional characteristics (current implementation) and have something like `BT_TBS_CLIENT_MINIMAL` that reverses the `default y` to `default n` (e.g. by doing `default !BT_TBS_CLIENT_MINIMAL`), and then enable the 2 Kconfigs. 

I do not think the solution of disabling 18 Kconfigs in order to just have 2 enabled is the right way to go.